scalyr_agent/sharded_copying_manager/copying_manager.py

```diff
@@ -149,62 +161,38 @@ def generate_status(self):
 
         result.api_key_id = self.__api_key_id
 
-        last_response_statuses = set()
-        health_checks = set()
-
-        last_attempt_requests_overall_size = 0
-
         for worker in sorted(self.__workers, key=lambda w: w.get_id()):
             status = worker.generate_status()
             result.workers.append(status)
 
             # get the most recent time fields from workers.
             if status.last_success_time:
-                result.last_success_time = _max_ignore_none(
+                result.last_success_time = max_ignore_none(
                     result.last_success_time, status.last_success_time
                 )
             if status.last_attempt_time:
-                result.last_attempt_time = _max_ignore_none(
+                result.last_attempt_time = max_ignore_none(
                     result.last_attempt_time, status.last_attempt_time
                 )
 
             if status.last_attempt_size:
-                last_attempt_requests_overall_size += status.last_attempt_size
+                result.last_attempt_requests_overall_size += status.last_attempt_size
 
             if status.last_response_status is not None:
-                last_response_statuses.add(status.last_response_status)
+                # do not overwrite the result if we already have a bad response.
+                if result.all_responses_successful is not False:
+                    result.all_responses_successful = status.last_response_status == "success"
 
-            if (
-                status.health_check_result is not None
-            ):
-                health_checks.add(status.health_check_result)
+            if status.health_check_result is not None:
+                # do not overwrite the result if there is a bad health check.
+                if result.all_health_checks_good is not False:
```

Comment:
Just write this as `not result.all_health_checks_good`.

Reply:
https://github.com/scalyr/scalyr-agent-2/pull/654#discussion_r528000579